pkgs/development/tools/gomod2nix/generic.nix

```diff
@@ -0,0 +1,45 @@
+{ version
+, src
+, modules
+}:
+{ stdenv
+, callPackage
+, go
+, lib
+, makeWrapper
+, installShellFiles
+, fetchFromGitHub
+, buildGoApplication ? (callPackage ./builder { }).buildGoApplication
+, mkGoEnv ? (callPackage ./builder { }).buildGoApplication
```

Comment:
This is not really a great interface. It often causes lots of confusion if a typo is done in the variable names and things just do not work. Could we refactor this to be less error prone?

Reply:
I've reworked it and just accepted the duplication and possible divergence in expressions between upstream and nixpkgs instead of trying to maximally share.

The code is much cleaner this way.